aws/resource_aws_ecs_task_definition.go

```diff
@@ -230,6 +230,56 @@ func resourceAwsEcsTaskDefinition() *schema.Resource {
 				}, false),
 			},
 
+			"proxy_configuration": {
+				Type:     schema.TypeList,
+				MaxItems: 1,
+				Optional: true,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"container_name": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+						"properties": {
+							Type:     schema.TypeSet,
```

Comment:
As mentioned below, we may want to consider using a map attribute here since its already string key and value instead of requiring configuration blocks. 😄 

```suggestion
							Type:     schema.TypeMap,
							Elem:     &schema.Schema{Type: schema.TypeString},
```

Along with the necessary updates below. 👍

Reply:
updated